livre/core/pipeline/FutureMap.h

```diff
@@ -0,0 +1,192 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _FutureMap_h_
+#define _FutureMap_h_
+
+#include <livre/core/types.h>
+#include <livre/core/pipeline/Future.h>
+
+namespace livre
+{
+
+/**
+ * FutureMap is a wrapper class to query multiple futures with same names for data and data state.
+ */
+class FutureMap
+{
+public:
+
+    /**
+     * @param futures is the list of futures
+     * future names are used for name-future association.
+     */
+    explicit FutureMap( const Futures& futures );
+    ~FutureMap();
+
+    /**
+     * Gets the copy of value(s) with the given type T. If input
+     * is connected and values are not provided this function will
+     * block.
+     * @param name of the future.
+     * @return the values of the futures.
+     * @throw std::runtime_error when the port data is not exact
```

Comment:
The first mention to a port shouldn't appear here. Mention how this class relates to ports in the class description.